e2e/consulcompat/shared_run_test.go

```diff
@@ -280,4 +280,8 @@ func runConnectJob(t *testing.T, nc *nomadapi.Client, ns, filePath string) {
 	// Ensure that the template rendered
 	_, _, err = nc.AllocFS().Stat(alloc, "dashboard/local/count-api.txt", nil)
 	must.NoError(t, err)
+
+	// Ensure that Consul tokens were written to task secret dir
+	_, _, err = nc.AllocFS().Stat(alloc, "dashboard/secrets/consul_token", nil)
+	must.NoError(t, err)
```

Comment:
It occurs to me that this won't pass for the legacy test, because we only write this token if there's a Consul identity.

Reply:
rrright. I moved it to `consul` directory above in [afdbe15](https://github.com/hashicorp/nomad/pull/19500/commits/afdbe1559a128ee6f70511b9808a04819ed8fbaa)